### ISSUE TITLE
Tutorials on the Wand...

### ISSUE THREAD

**Issue by Talgoni** [maruohon/enderutilities#96]
I've been looking for days now for a tutorial on using the Lazy wand...
everyone is complimentary on it and there's a bunch of demostrating, but not a single person is actually telling HOW to use it, as in what buttons/keys to press. Specifically...

1) I can't find the button that actually STARTS actions that have 2 points of reference, it doesn't appear to be in the tooltips.
( single point is fine, my right mouse button works & off we go BUT 'Delete' for example uses 2 points & clicking right mouse there just moves the 2nd point. )

2) Also; 'Activate Unselected'? - seems to do nothing & there's no info?

3) You use Shift & Sneak in the tooltip...they're the same key to me, but if I remapped Sneak onto a different key do you still mean Shift & Sneak to be different?
4) Have you considered an ingame manual?

It would be nice if the tutorials displayed the keys/buttons in use even if they don't say what to use.
I can't seem to find the kletsie wand spotlight.

Plaese help,
Thanks.
:-)

**Comment by maruohon** [comment]
The wand is quite the usability hell for sure... and lots of it stems from the early design decision of not using more than one main hotkey in the mod, although later I added the second "unselected" variant too, although I think I usually kept them in the same key.

The two-point modes are activated by holding down right click for a couple of seconds while not looking at any blocks (like a bow charging). Granted, this is rather annoying if you are in a tight space where it's not possible to not look at anything... >_>

The Activate Unselected is basically like the normal Toggle key, but it works also for items that you are not holding in your hand. Mainly it's useful for things like the Inventory Swapper, to enable the swapping operation while having the item somewhere other than your hotbar.

The Shift vs. Sneak terminology might be mixed up in the tooltips, I can't remember really... it has been a rather long time since I last actively worked on the mod. I don't remember how many things would check for the actual shift key.

A manual is on my TODO-list, and has been there for several years... you know how it sometimes goes with trying to write documentation for things >_>

The wand in particular would need a complete usability overhaul. I was planning on rewriting it, or adding a second version of it, or making a separate mod with the improved version... But I've been busy with other mods lately, plus then Building Gadgets was released at some point and thus the potential users of an improved wand in any of my mods kinda dropped to minimum most likely. So I kinda abandoned that project before ever even starting on it. It is still on my ever-growing TODO-list, so if I ever run out of other things to work on, I might re-visit that idea. Plus many of the things I've worked on for Litematica recently would apply/fit in that mod as well.

**Comment by Talgoni** [comment]
Thanks for the quick response! :-)
I was prompted to post because I've recently been doing Sky Factory 4 and, of course, it's included in there along with quests such as 'build a 24x24x24 Nuclearcraft reactor'!

The bow-charging helps a LOT!!! :-D 
Worked perfectly and means I can got things done at last!
Would it be worth just a minor tooltip edit releease?
I, for one, would love to see a new version, but, yep, I know about that 'life' stuff that gets in the way! LOL

**Comment by IngmarBitter** [comment]
The main feature I am looking for is MOVE, so if i want to enlarge a house I can move walls out.
I just made the wand (with lots of endgame item effort) in SevTech ages (MC1.12.2) and was excited to reorganize my base with the wand.
 
MOVE has a two point volume of interest. I seem to only be able to set one point with left click. Right click does noting. (Except for holding long, then it executes the move.)
How do I clean the volume and start over?

Please explain.
Thanks!

**Comment by maruohon** [comment]
(wow the usability of the wand is super terrible, I couldn't even remember how anything works myself and what the magic keybind combos are without looking at the tooltip...)

Okay so the Move feature has two different sub-modes, `Move, from` where you select the area to move by left and right clicking (and shift + scrolling) to place the corners, and the `Move, to` mode where you position the paste target area by just left clicking to set the primary corner and optionally rotating and/or mirroring the area, and then you hold right click on air and release to execute the move operation.

**Comment by IngmarBitter** [comment]
Thank you for the fast response!

> select the area to move by left and right clicking
That means if I left click on a block and get the selection box there (in red) and then right click on the block next to it then I should get a blue box there? If the blue box does not move then the right click selection is broken?

If that was working then I thin the usability is not bad at all.  It's in the nature of the beast so to speak that 3d selection is a little more complicated.  And if right click sets the other volume of interest corner than its quite intuitive.

**Comment by maruohon** [comment]
Right clicking should get you the blue corner, yes. Make sure you are in the `Move, from` mode when selecting what to move. And then in the `Move, to` mode you select where to move that first selection area.

**Comment by IngmarBitter** [comment]
Yes checked those modes.  I can circumvent the bug via shift+scroll.
Also, resetting to not rotated is not clear how to do.

**Comment by IngmarBitter** [comment]
if the wand operations can be represented with icons, then having a cheat cheat with the icons and how they relate to the keyboard and mouse would make it easier to see how it can be used. Here is an example that is using icons form a different application domain, but I think it makes the idea clear:
![image](https://user-images.githubusercontent.com/4473591/80177702-6a357e00-85ca-11ea-83a8-4553b0f37a57.png)

**Comment by maruohon** [comment]
Oh wow that looks pretty cool.

But I *really* should just add a GUI you can open to set the per-mode settings instead of using up to 8 obscure keybind combinations. Those just developed as part of the early design decision to only use one main keybind in the mod (because the amount of keybinds and available keys for those in large modpacks was traditionally a huge issue. And nowadays EnU already has two keybinds... although they are almost the same thing, the second just works for items that are not in your hand) so then I needed modifier combinations to achieve multiple different actions with that.

I was planning on adding another variant of the wand that would basically just be completely rewritten from scratch both code-wise and usability-wise. But then the Building Gadgets mod was released at some point and I thought that nobody would bother using my mod for any building stuff anymore anyway and I always had other mods to work on anyway so I never ended up working on that newer wand version (at least thus far...). And I was also considering making that as a separate mod, which is one reason why Building Gadgets kinda trumps it as the already established popular alternative for the same kind of task.

**Comment by IngmarBitter** [comment]
To me _move_ is actually the most valuable of all the modes.

I did not try the building Building Gadgets yet. They also need a ton of endgame parts to make, and have no in-game description at all.  There are three of them. The _Copy Paste Gadget_ sounds like it might be able to do _move_ as well?  

I could also offer to debug and fix the right click not working issue. I have quite some programming experience, but a short call on discord to get the dev environment setup would be nice. I can setup connecting to and cloning the GIT repository before that.
